crates/guest/assessor/assessor-guest/src/main.rs

```diff
@@ -64,23 +61,30 @@ fn main() {
             fill.verify_signature(&eip_domain_separator).expect("signature does not verify")
         };
         fill.evaluate_requirements().expect("requirements not met");
-        env::verify_integrity(&fill.receipt_claim()).expect("claim integrity check failed");
-        let claim_digest = fill.receipt_claim().digest();
+        let claim_digest = fill.claim_digest().expect("failed to get claim digest");
         let commit = AssessorCommitment {
             index: U256::from(index),
             id: fill.request.id,
             requestDigest: request_digest.into(),
             claimDigest: <[u8; 32]>::from(claim_digest).into(),
+            fulfillmentDataDigest: <[u8; 32]>::from(fill.fulfillment_data_digest()).into(),
         }
         .eip712_hash_struct();
         leaves.push(Digest::from_bytes(*commit));
+
+        let callback = &fill.request.requirements.callback;
+
+        // Note that this allows for callbacks to be specified even in the case of
+        // ClaimDigestMatch predicates where the journal cannot be provided.
+        // This is ok because it will fail on chain checks if a journal is indeed provided.
```

Comment:
This is not quite accurate, because it _is_ possible for a prover to provide the image ID and journal fulfillment data in response to a `ClaimDigestMatch` request. In this case, if a callback was requested as well, it will be delivered. This is all correct behavior, even though we don't expect provers to actually do this.

Reply:
Oops i left an old comment there